### DISCUSSION TITLE
SetScreen funciona en android pero no en Desktop

### DISCUSSION THREAD

**Discussion by jotayc** [danirod/jumpdontdie#34] [Q&A]
Buenas,

   siguiendo tu tutorial, y haciendo pruebas tanto en tu juego como en el mío, me surge una duda tras comprobar que en ambos da un error. Y es que, una vez el personaje choca con algo y debemos pasar a la ventana de GameOver, pones una secuencia de acciones con un delay de 1.5f, y posteriormente cambias de escena.  Justo así: 

![image](https://user-images.githubusercontent.com/7070284/154272543-88536e1a-2e4c-4ac5-8170-f0402c6d59bb.png)

Me preguntaba qué pasaría si cambias de pantalla directamente, sin necesidad de hacer la secuencia de acciones. Así: 
![image](https://user-images.githubusercontent.com/7070284/154272839-4c27ff85-9d6a-47e0-b977-1fa756fdaedf.png)

La cuestión es que no sé porqué, al ejecutarlo de esta segunda forma en Desktop la aplicación da error, mientras que en Android no. 

No se si sabrías decirme porqué, y si ese delay que pones va en este sentido.

Gracias, 
Un saludo.

**Comment by danirod** [comment]
No puedo decir mucho porque hace bastante que no reviso este código, pero es posible que esto esté pasando porque sin el delay, que realmente está metido por temas cosméticos, estarías cambiando de pantalla dentro de uno de los handlers de Box2D. O sea que lo mismo está provocando que se llame la hide() del GameScreen en un momento en el que Box2D todavía no ha terminado de hacer sus cosas.
